Lesson15/nurtap/script.js

```diff
@@ -0,0 +1,390 @@
+      newArr.push(element);
+    }
+  }
+  console.log(newArr);
+}
+removeDuplicates([5, 3, 8, 5, 7, 8]);
+
+/*
+13. Calculate Factorial (For Loop)
+    - Define a function `factorial(n)` that calculates n! (n factorial)
+      using a for loop.
+    - Log the result. 
+    - Example: factorial(5) -> 120
+*/
+
+function factorial(n) {
+  let result = 1;
+  for (let i = 1; i <= n; i++) {
+    result *= i;
+  }
+  console.log(result);
+}
+factorial(5);
+
+/*
+14. String -> Array -> String
+    - Define a function `reverseWords(sentence)` that splits the sentence 
+      into an array of words, reverses the array order, then joins it back into
+      a string. Use loops or built-in methods as you like.
+    - Log the reversed sentence.
+*/
+
+function reverseWords(sentence) {
+  console.log(sentence.split(" ").reverse().join(" "));
+}
+reverseWords("life is good");
+
+/*
+15. Filter Words Longer Than X
+    - Define a function `filterLongWords(words, minLength)` that uses a for loop
+      to collect only the words that have a length >= minLength.
+    - Log the resulting array.
+*/
+
+function filterLongWords(words, minLength) {
+  let filteredWords = [];
+  for (let word of words) {
+    if (word.length >= minLength) {
+      filteredWords.push(word);
+    }
+  }
+  console.log(filteredWords);
+}
+
+filterLongWords(["nurtap", "jan", "ramazan"], 6);
+
+/*
+16. Log Array Elements with Their Indices
+    - Define a function `logElementsWithIndex(arr)` that loops through the array
+      and logs "Index: i, Value: arr[i]" for each element.
+*/
+
+function logElementsWithIndex(arr) {
+  for (let i = 0; i < arr.length; i++) {
+    console.log(`Index: ${i}, Value: ${arr[i]}`);
+  }
+}
+logElementsWithIndex([2, 4, 6, 8]);
+
+/*
+17. Find the Smallest Number in an Array
+    - Define a function `findMin(numbers)` that loops through the array
+      to find and return the smallest number.
+    - Log the smallest number.
+*/
+function findMin(numbers) {
+  let smallestNumber = numbers[0];
+  for (let number of numbers) {
+    if (number < smallestNumber) {
+      smallestNumber = number;
+    }
+  }
+  console.log(smallestNumber);
+}
+
+findMin([5, 8, 2, 7]);
+
+/*
+18. Count Occurrences of a Word in an Array
+    - Define a function `countOccurrences(arr, word)` that loops through `arr`
+      to count how many times `word` appears.
+    - Log the count.
+*/
+
+function countOccurrences(arr, word) {
+  let count = 0;
+  for (let elem of arr) {
+    if (elem == word) {
+      count += 1;
+    }
+  }
+  console.log(count);
+}
+
+countOccurrences(["nurtap", "serhat", "ece", "nurtap", "sevtap"], "nurtap");
+
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+function removeFalsyValues(arr) {
+  let newArr = [];
+  for (let elem of arr) {
+    if (!elem) {
```

Comment:
Use `if (elem)` instead of `if (!elem)` to keep the truthy value:
```suggestion
    if (elem) {
```